File: 0001_0100/0046/0046.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.read())
    s = "2.7182818284590452353602875"
    if n < 25:
        s = "3" if n == 0 else s[:n + 1] + str(int(s[n + 1]) + 1) if s[n + 2] >= '5' else s[:n + 2]
    q.write(s)

File: 0001_0100/0046/0046.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n int
	s := "2.7182818284590452353602875"
	fmt.Fscan(f, &n)
	if n == 0 {
		s = "3"
	} else if n < 25 {
		if s[n+2] >= '5' {
			s = fmt.Sprintf("%s%c", s[:n+1], s[n+1]+1)
		} else {
			s = s[:n+2]
		}
	}
	fmt.Fprint(q, s)
}
